cli/internal/terraform/terraform/azure/modules/scale_set/variables.tf

```diff
@@ -1,7 +1,31 @@
-variable "name" {
+variable "base_name" {
   type        = string
-  default     = "constell"
-  description = "Base name of the cluster."
+  description = "Base name of the instance group."
+}
+
+variable "node_group_name" {
+  type        = string
+  description = "Constellation name for the node group (used for configuration and CSP-independent naming)."
+}
+
+variable "role" {
+  type        = string
+  description = "The role of the instance group."
+  validation {
+    condition     = contains(["ControlPlane", "Worker"], var.role)
+    error_message = "The role has to be 'ControlPlane' or 'Worker'."
+  }
+}
+
+variable "tags" {
+  type        = map(string)
+  description = "Cluster tag."
```

Comment:
In the description you use `tag` as singular while the variable name and type signal that this (can) contain multiple `tags`.